File: problems/detec_loop_Linklist.py
# User function Template for python3
'''
        # Node Class
        class Node:
            def __init__(self, data):   # data -> value stored in node
                self.data = data
                self.next = None
    
'''


class Solution:
    # Function to check if the linked list has a loop.
    def detectLoop(self, head):
        # code here

        slow = head
        fast = head

        while fast and fast.next:
            # slow jumps one step further

            # fast jumps two steps forward

            slow = slow.next
            fast = fast.next.next
            # At some point fast meets slow if loop is found in linked list
            if fast == slow:
                return True
        return False

File: LinkList/detec_loop_link_list.cpp
// User function template for C++

/*

struct Node
{
    int data;
    struct Node *next;
    Node(int x) {
        data = x;
        next = NULL;
    }

*/
class Solution
{
public:
     // Function to check if the linked list has a loop.
     bool detectLoop(Node *head)
     {
          // your code here
          if (head == NULL || head->next == NULL)
               return false;

          Node *fast = head;
          Node *slow = head;

          while (slow && fast && fast->next and fast->next->next)
          {
               fast = fast->next->next;
               slow = slow->next;
               if (fast == slow)
                    return true;
          }
          return false;
     }
};